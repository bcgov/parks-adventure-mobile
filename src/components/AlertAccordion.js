import React from 'react'
import PropTypes from 'prop-types'
import HTML from 'react-native-render-html'
import { AllHtmlEntities } from 'html-entities'
import { useTheme } from 'react-native-paper'
import { MaterialCommunityIcons as Icon } from '@expo/vector-icons'
import {
  Container,
  Accordion,
  Content,
  Headline,
  HeadlineText,
} from './AlertAccordion.styles'

const entities = new AllHtmlEntities()

function AlertAccordion({ headline, description, alert = false }) {
  const theme = useTheme()
  const [expanded, setExpanded] = React.useState(false)

  const containsDesciption = description && description.match(/\w/)
  const color = alert ? 'white' : 'black'

  return (
    <Container>
      {containsDesciption ? (
        <Accordion
          testID={'accordion'}
          alert={alert}
          title={entities.decode(headline)}
          expanded={expanded}
          onPress={() => setExpanded(!expanded)}
          titleNumberOfLines={3}
          titleStyle={{
            color,
            fontSize: 14,
            fontFamily: alert ? 'bcsans-bold' : 'bcsans',
          }}>
          <Content alert={alert}>
            <HTML
              html={`<div class='description'>${description}</div>`}
              classesStyles={{ description: { color } }}
              tagsStyles={{
                a: { color: theme.colors.primary },
                li: { paddingTop: 5 },
              }}
              listsPrefixesRenderers={{
                ul: () => {
                  return (
                    <Icon
                      name="circle-medium"
                      color={color}
                      size={12}
                      style={{ paddingTop: 7 }}
                    />
                  )
                },
              }}
            />
          </Content>
        </Accordion>
      ) : (
        <Headline>
          <HeadlineText>{headline}</HeadlineText>
        </Headline>
      )}
    </Container>
  )
}

AlertAccordion.propTypes = {
  headline: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  alert: PropTypes.bool,
}

export default AlertAccordion
