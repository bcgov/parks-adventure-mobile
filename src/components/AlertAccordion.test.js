import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import renderSnapshot from 'react-test-renderer'
import AlertAccordion from './AlertAccordion'

test('Should render alert', () => {
  const { getByText } = render(
    <AlertAccordion
      headline="This is the title"
      description="This is information about the alert"
      type={'alert'}
    />
  )

  const title = getByText('This is the title')
  expect(title).toBeDefined()

  fireEvent.press(title)

  const description = getByText('This is information about the alert')
  expect(description).toBeDefined()
})

test('Should render headline without accordion if no advisory desciption', () => {
  const { queryByTestId } = render(
    <AlertAccordion headline="Title" description=" " type={'alert'} />
  )

  const accordion = queryByTestId('accordion')
  expect(accordion).toBeNull()
})

test('matches alert snapshot', () => {
  const tree = renderSnapshot
    .create(
      <AlertAccordion
        headline="Title"
        description="Description"
        type={'alert'}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('matches advisory snapshot', () => {
  const tree = renderSnapshot
    .create(
      <AlertAccordion
        headline="Title"
        description="Description"
        type={'advisory'}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
