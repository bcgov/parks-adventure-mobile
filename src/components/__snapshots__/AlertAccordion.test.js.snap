// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches advisory snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "elevation": 4,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
    }
  }
>
  <View>
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Array [
            Object {
              "padding": 8,
            },
            Array [
              Object {
                "backgroundColor": "#FADA85",
                "color": "black",
              },
            ],
          ],
        ]
      }
      testID="accordion"
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={3}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                  },
                  Object {
                    "color": "black",
                    "fontFamily": "bcsans",
                    "fontSize": 14,
                  },
                ],
              ]
            }
          >
            Title
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "material-community",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`matches alert snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "elevation": 4,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
    }
  }
>
  <View>
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Array [
            Object {
              "padding": 8,
            },
            Array [
              Object {
                "backgroundColor": "#FF0C3E",
                "color": "white",
              },
            ],
          ],
        ]
      }
      testID="accordion"
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={3}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                  },
                  Object {
                    "color": "white",
                    "fontFamily": "bcsans-bold",
                    "fontSize": 14,
                  },
                ],
              ]
            }
          >
            Title
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "material-community",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
